src/viewport.h

```diff
@@ -55,7 +55,7 @@ struct viewport {
     enum vp_scale def_scale;  ///< Default image scale
 
     double scale;         ///< Scale factor of the image
-    ssize_t x, y;         ///< Image position on the window surface
+    double x, y;          ///< Image position on the window surface
```

Comment:
Looks very strange now.
The image position is tied to the window, i.e. to pixels, and must be stored as an integer value.

Reply:
> Tracking the image position to a sub-pixel precision

I agree that this is a good idea, but then we also need to properly handle double precision math, for example [here](https://github.com/artemsen/swayimg/blob/6b3539b1c312726ea4317bd02ae99d21569ee2ce/src/viewport.c#L76) and [here](https://github.com/artemsen/swayimg/blob/master/src/viewport.c#L350). We need to change the interface of some functions that still use `size_t`, such as [viewport_move](https://github.com/artemsen/swayimg/blob/6b3539b1c312726ea4317bd02ae99d21569ee2ce/src/viewport.c#L325C6-L325C19).

So if doubles ​​are really necessary, let's do it in another PR, because checking all the corner cases might take some time.